batch/batch/batch.py

```diff
@@ -284,28 +288,40 @@ class Job:
         assert self._state in ('Pending', 'Cancelled')
         return None
 
-    async def _mark_job_task_complete(self, task_name, log, exit_code, new_state, pod_status):
-        self.exit_codes[self._task_idx] = exit_code
-        self.pod_statuses[self._task_idx] = pod_status
-
+    async def _mark_job_task_complete(self, task_name, pod_log, exit_code, new_state, pod_status):
         uri = None
-        if log is not None:
-            uri, err = await app['log_store'].write_gs_log_file(*self.id, task_name, log)
+        if pod_log is not None:
+            uri, err = await app['log_store'].write_gs_log_file(*self.id, task_name, pod_log)
```

Comment:
needs a random number on file name in case this is the second time the task was run.

Reply:
this is a harder change to make and not essential